base/macros.hpp

```diff
@@ -163,6 +163,9 @@ inline void noreturn() { std::exit(0); }
 #  error "What compiler is this?"
 #endif
 
+// We assume that the processor is at least a Nehalem.
```

Comment:
Rename to PRINCIPIA_USE_SSE3_INTRINSICS, use pmmintrin (we only use SSE2, which means we require a Pentium 4, but we also require x86-64, so we actually require Prescott or newer).

Reply:
Done.